lib/widgets/widgets/zabbix/zabbix_widget.dart

```diff
@@ -0,0 +1,381 @@
+import 'package:cogboardmobileapp/models/zabbix_chart_item_model.dart';
+import 'package:cogboardmobileapp/widgets/widgets/zabbix/zabbix_chart.dart';
+import 'package:charts_flutter/flutter.dart' as charts;
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class ZabbixWidget extends StatefulWidget {
+  final DashboardWidget widget;
+
+  ZabbixWidget({
+    @required this.widget,
+  });
+
+  @override
+  _ZabbixWidgetState createState() => _ZabbixWidgetState();
+}
+
+class _ZabbixWidgetState extends State<ZabbixWidget> with SingleTickerProviderStateMixin {
+  int previousValue;
+  AnimationController _controller;
+  Tween<double> valueTween;
+
+  @override
+  void initState() {
+    super.initState();
+    previousValue = int.parse(widget.widget.content['lastvalue']);
+    this.valueTween = Tween(
+      begin: calculatePercentageValue(int.parse(getLastValue)) / 100,
+      end: calculatePercentageValue(int.parse(getLastValue)) / 100,
+    );
+    this._controller = AnimationController(
+      duration: const Duration(milliseconds: 1000),
+      vsync: this,
+    );
+    this._controller.forward();
+  }
+
+  @override
+  void dispose() {
+    this._controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(ZabbixWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (previousValue != int.parse(getLastValue)) {
+      double begin = this.valueTween?.evaluate(this._controller) ?? calculatePercentageValue(previousValue) ?? 0;
+      this.valueTween = Tween<double>(
+        begin: begin,
+        end: calculatePercentageValue(int.parse(getLastValue)) / 100,
+      );
+      this._controller
+        ..value = 0
+        ..forward();
+    }
+  }
+
+  List get getHistory {
+    Map historyMap = widget.widget.content['history'];
+    return historyMap.entries
+        .map((e) => ZabbixHistoryItem(
+              timestamp: e.key,
+              value: e.value,
+            ))
+        .toList();
+  }
+
+  bool get displayUpArrow {
+    List historyData = getHistory;
+    int current = int.parse(historyData[historyData.length - 1].value);
+    int prev = int.parse(historyData[historyData.length - 2].value);
+    return current > prev;
+  }
+
+  String get getTitle {
+    return ZabbixMetrics[widget.widget.selectedZabbixMetric];
+  }
+
+  int get getMaxValue {
+    return widget.widget.maxValue;
+  }
+
+  int get rangeLowerBound {
+    return widget.widget.range[0];
+  }
+
+  int get rangeUpperBound {
+    return widget.widget.range[1];
+  }
+
+  String get getProgressType {
+    return checkMetricHasProgress() && (!checkMetricHasMaxValue() || widget.widget.maxValue > 0)
+        ? 'progress'
+        : 'maxValue';
+  }
+
+  bool get isSystemUptime {
+    return widget.widget.content['name'] == "System uptime";
+  }
+
+  String get getLastValue {
+    return widget.widget.content['lastvalue'];
+  }
+
+  bool checkMetricHasMaxValue() {
```

Comment:
[LATER] change method name to something more meaningful like: `isInBytes()`

Reply:
done